GHzDACs/ghz_fpga_server.py

```diff
@@ -1855,8 +1963,46 @@ def dac_bist(self, c, chan, data):
         cmd, shift = dac.DAC.getCommand({'A': (2, 0), 'B': (3, 14)}, chan)
         dev = self.selectedDAC(c)
         ans = yield dev.runBIST(cmd, shift, data)
-        returnValue(ans)
-    
+        # This is coming back with 64-bit ints, the coercing of which needs to be fixed in pylabrad
+        # for now we manually cast to 32-bit (long)
+        # See pylabrad github issue #43
+        returnValue((ans[0],
+                    [long(x) for x in ans[1]], [long(x) for x in ans[2]], [long(x) for x in ans[3]],
+                     ))
+
+    @setting(1301, 'DAC Bringup New', lvdsOptimize='b', lvdsSD='w', signed='b', targetFifo='w',
+             returns='*(bs)')
+    def dac_bringup(self, c, lvdsOptimize=False, lvdsSD=None, signed=True, targetFifo=None):
+        """
+
+        :param c:
+        :param lvdsOptimize:
+        :param lvdsSD:
+        :param signed:
+        :param targetFifo:
+        :return: list of (bool, string) tuples indicating success of each test
+        """
+        dev = self.selectedDAC(c)
+        yield dev.initPLL()
+        time.sleep(0.100)
+        yield dev.resetPLL()
+        results = []
+        for dac, cmd, shift in zip(['A', 'B'], (2, 3), (0, 14)):
+            ansDAC = [('dac', dac)]
+            #Initialize DAC
+            #See HardRegProgram.txt for byte sequence definition
+            pkt = [0x0024, 0x0004, 0x1603, 0x0500] if signed else \
+                  [0x0026, 0x0006, 0x1603, 0x0500]
+            yield dev.runSerial(cmd, pkt)
+            lvdsAns = yield dev.setLVDS(cmd, lvdsSD, lvdsOptimize)
+            results.append((lvdsAns[0], "setLVDS"))
+            fifoAns = yield dev.setFIFO(dac, cmd, targetFifo)
+            results.append((fifoAns[0], "setFIFO"))  # TODO: do we want to keep num. tries as a result?
```

Comment:
I vote for returning the number of FIFO tries (since this indicates how close we are to a FIFO failure even if there is no failure. (Note: failure occurs after 5 unsuccessful tries). Perhaps save this as a parameter that gets appended to results after the conclusion of the for loop (so you don't have ints breaking up your nice concise return type tag).